### PR TITLE
fix(crud): If a date is in the safe range, go with relaxed EJSON rather than canonical COMPASS-5744

### PR DESCRIPTION
**Pull request by lerouxb**
Some reference:

https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#datetime
https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#conversion-table
https://github.com/mongodb/js-bson/blob/35bb2a279343439e1306a0eeb8d65d1abf910929/test/node/specs/bson-corpus/datetime.json#L9-L10

### CODE REVIEW

File: packages/hadron-document/src/utils.ts [hunk 1]
```diff
@@ -70,6 +76,12 @@ function makeEJSONIdiomatic(value: EJSON.SerializableTypes): void {
       }
       continue;
     }
+    if (entry.$date && entry.$date.$numberLong) {
+      const number = entry.$date.$numberLong;
+      if (number >= 0 && number <= maxFourYearDate) {
+        entry.$date = new Date(+number);
```
Comment (addaleax) [inline packages/hadron-document/src/utils.ts:82]: Should this be

```suggestion
        entry.$date = new Date(+number).toISOString();
```

?

Comment (addaleax) [reply packages/hadron-document/src/utils.ts:82]: (also, you may want to add tests for this)

Comment (lerouxb) [reply packages/hadron-document/src/utils.ts:82]: I don't know. It seems to work. I suppose I can just add that to be safe.

Comment (lerouxb) [reply packages/hadron-document/src/utils.ts]: Yeah I see this file has no tests yet :(

Comment (addaleax) [reply packages/hadron-document/src/utils.ts]: It does: https://github.com/mongodb-js/compass/blob/03211164e94cb45075ead41b2090cacd99a71301/packages/hadron-document/test/document.test.ts#L2317-L2382

Comment (lerouxb) [reply packages/hadron-document/src/utils.ts]: Added now.

**Review by addaleax — APPROVED** [review]

**Pull request merged**